### DISCUSSION TITLE
Then chaining base on condition

### DISCUSSION THREAD

**Discussion by zorture** [mxcl/PromiseKit#1207] [Q&A]
Is the below code possible? I am getting Error.  Type '()' cannot conform to 'Thenable'; only struct/enum/class types can conform to protocols.
How do I achieve condition chaining inside then closure?

        firstly {
            Promise.value(1)
        }.then { _ in
            Promise.value(2)
        }.then { value in
            if value > 0 {
                Promise.value(3)
            } else {
                Promise.value(4)
            }
        }.done {
            print($0)  
        }

**Comment by GarthSnyder** [comment]
`if` in Swift is not an expression, so the last `then` returns nothing, aka Void, aka (). Just put explicit returns in for values 3 and 4.

**Comment by mxcl** [comment]
Probs need to specify return type too:

```swift
firstly {
    Promise.value(1)
}.then { _ in
    Promise.value(2)
}.then { value -> Promise<Int> in
    if value > 0 {
        return Promise.value(3)
    } else {
        return Promise.value(4)
    }
}.done {
    print($0)  
}
```
